code/game/say.dm

```diff
@@ -76,13 +76,35 @@ var/list/freqtoname = list(
 	say_testing(src, "/atom/movable/proc/send_speech() start, msg = [speech.message]; message_range = [range]; language = [speech.language ? speech.language.name : "None"];")
 	if(isnull(range))
 		range = 7
+	range = atmospheric_speech(speech,range)
 	var/rendered = render_speech(speech)
 	var/list/listeners = get_hearers_in_view(range, src)
 	if(speech.speaker.GhostsAlwaysHear())
 		listeners |= observers
 	for(var/atom/movable/AM in listeners)
 		AM.Hear(speech, rendered)
 
+/atom/movable/proc/atmospheric_speech(var/datum/speech/speech, var/range=7)
+	var/turf/T = get_turf(speech.speaker)
```

Comment:
Note to self to change this, it doesn't factor in the possibility of being somewhere with its own air environment  (notably a space pod or inflatable shelter)

Reply:
Do you want to do it with this PR?

Reply:
Yes, sorry about leaving this one without work. Ive had NO INTERNET for 5 days and all my shitposting has been on borrowed networks.

I will have changes up this week, promise.

Reply:
you lied

Reply:
It turns out GFs are really good at finding ways to make you not code

Reply:
normie. Do the thing!